src/Avalonia.Controls/IndexRange.cs

```diff
@@ -3,12 +3,17 @@
 //
 // Licensed to The Avalonia Project under MIT License, courtesy of The .NET Foundation.
 
+using System;
+using System.Collections.Generic;
+
 #nullable enable
 
 namespace Avalonia.Controls
 {
-    internal readonly struct IndexRange
+    internal readonly struct IndexRange : IEquatable<IndexRange>
     {
+        private static readonly IndexRange s_invalid = new IndexRange(int.MinValue, int.MinValue);
```

Comment:
>s_invalid [](start = 43, length = 9)

Curious why this is needed? Can we just not have that index Range instead ? Or is just the initial state ?

Reply:
Sorry, don't understand, could we not just have what index Range?

Reply:
is s_invalid needed ? when is a range invalid ? An empty range makes sense, but what case does a range become invalid ?

Reply:
Got you. Yeah, it's only used internally as a [sentinal value in `Add()`](https://github.com/AvaloniaUI/Avalonia/blob/88794773229e5c9fe93f29672c41b9cbac601a89/src/Avalonia.Controls/IndexRange.cs#L97) to indicate that we're finished adding the range. Adding a separate bool flag would work as well, but I figured that we have a struct with 8,589,934,590 possible values so I might as well use one of those values to indicate that the loop is finished... I assumed that it's unlikely that a range of `[-minvalue..-minvalue]` would be used in a real situation.